### PR TITLE
Readme add generators

### PR DESCRIPTION
**Pull request by kalabYibeltal**

### CODE REVIEW

**Comment by saicharan0112** [comment]
https://github.com/idea-fasoc/OpenFASOC/pull/46#issuecomment-1116121744 @kalabYibeltal Your pre-commit test is failing. Follow this comment and push changes to your repo. And please do not close this PR. You can directly push commits to this PR.

**Comment by kalabYibeltal** [comment]
@saicharan0112 How do I do that without closing the PR?

**Comment by saicharan0112** [comment]
@kalabYibeltal I didn't get you. Why are you referring that issue here?
You can just add changes to your fork and that will be reflected in your PR automatically if you keep it open. But it is up to you.

**Comment by msaligane** [comment]
@kalabYibeltal 
you need to install `pre-commit install` before `git commit`

if you have commited some code already that the formatter does not like you need to run `pre-commit run --all-files`

**Comment by saicharan0112** [comment]
@kalabYibeltal  run these two commands in after the other - 

- `pip install pre-commit`
- `pre-commit run --all-files` 

This will adjust your file. Now just commit the changes to the repository. Since you already have the PR open, those commits will reflect here. Once all the checks pass on that latest commit, it can be merged to this repository.

**Comment by kalabYibeltal** [comment]
@saicharan0112  I had to resolve dependency issues when installing the pre-commit package. Fortunately that has been fixed and I have run the pre-commit tests successfully locally.

File: README.md [hunk 1]
```diff
@@ -67,15 +67,33 @@ make sky130hd_temp
...
+
+  Block Architecture:
+   - Synchronous Digital LDO with optional stochastic flash ADC.
+    ![plot](./markdown%20images/LDO-BA.png)
+    Documentation **[link](https://fasoc.engin.umich.edu/digital-ldo/)**
+
+**[dcdc-gen:-](https://github.com/idea-fasoc/OpenFASOC/tree/main/openfasoc/generators/dcdc-gen)** For synthesizable on-chip power management  circuits, we use the “2:1 SC Cell” auxiliary cell for implementing a switched-capacitor (SC) based DC-DC converter. By varying the number of auxiliary cells, we can achieve wide range of conversion ratios with fine grain resolution. It operates similarly to  a successive approximation analog to digital converter (SAR ADC). Furthermore, since the total structure is simply composed of auxiliary cells, it is ideal for the proposed flow of automating the analog block design.  Documentation **[link](https://fasoc.engin.umich.edu/dc-dc-converter/)**
```
Comment (msaligane) [inline README.md:88]: I think the content of these links are not up to date. So don't point to them.

File: README.md [hunk 2]
```diff
@@ -67,15 +67,33 @@ make sky130hd_temp
 Please contact mehdi@umich.edu if you have any questions.
 
 
-#Generators
+# Generators
+
+**[temp-sense-gen:-](https://github.com/idea-fasoc/OpenFASOC/tree/main/openfasoc/generators/temp-sense-gen)** A fully automated SoC generator that uses an all-digital temperature sensor architecture, that relies on a new subthreshold oscillator (achieved using the auxiliary cell “Header Cell“) for realizing synthesizable thermal sensors.
```
Comment (msaligane) [inline README.md:72]: Can you add a link to a readme on each of the generators instead of having them on the main page?

File: README.md [hunk 3]
```diff
@@ -67,15 +67,33 @@ make sky130hd_temp
...
 
-**[temp-sense-gen:-](https://github.com/idea-fasoc/OpenFASOC/tree/main/openfasoc/generators/dcdc-gen)** A fully automated SoC generator that uses an all-digital temperature sensor architecture, that relies on a new subthreshold oscillator (achieved using the auxiliary cell “Header Cell“) for realizing synthesizable thermal sensors.
 **[cryo-gen:-](https://github.com/idea-fasoc/OpenFASOC/tree/main/openfasoc/generators/cryo-gen)**
-**[dcdc-gen:-](https://github.com/idea-fasoc/OpenFASOC/tree/main/openfasoc/generators/dcdc-gen)**
-**[gdsfactory:-](https://github.com/idea-fasoc/OpenFASOC/tree/main/openfasoc/generators/dcdc-gen)**
-**[lc-dco:-](https://github.com/idea-fasoc/OpenFASOC/tree/main/openfasoc/generators/dcdc-gen)**
-**[ido-gen:-](https://github.com/idea-fasoc/OpenFASOC/tree/main/openfasoc/generators/dcdc-gen)**
-**[scpa-gen:-](https://github.com/idea-fasoc/OpenFASOC/tree/main/openfasoc/generators/dcdc-gen)**
+
+**[gdsfactory:-](https://github.com/idea-fasoc/OpenFASOC/tree/main/openfasoc/generators/gdsfactory)**
```
Comment (msaligane) [inline README.md:92]: There is some documentation for these structures.

**Review by msaligane — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (kalabYibeltal) [reply README.md:72]: I couldn't find the readme file for any of the generators in their main pages or other folders.

File: README.md (hunk 3 above)
Comment (kalabYibeltal) [reply README.md:92]: I couldn't find documentation in https://fasoc.engin.umich.edu/ for the other four generators is there some other documentations I could look for?

File: README.md (hunk 2 above)
Comment (msaligane) [reply README.md:72]: you need to create one for each and put the info there

File: README.md (hunk 3 above)
Comment (msaligane) [reply README.md:92]: Check on slack with Tuohang.

**Comment by msaligane** [comment]
@kalabYibeltal are you addressing these things?

**Comment by kalabYibeltal** [comment]
> @kalabYibeltal are you addressing these things?

Yes, I wanted to add readme files for the generators then edit the hyperlinks. or should I I delete the links until I am done with the readmes?

**Comment by msaligane** [comment]
> > @kalabYibeltal are you addressing these things?
> 
> Yes, I wanted to add readme files for the generators then edit the hyperlinks. or should I I delete the links until I am done with the readmes?

Up to you!

**Comment by irakobi** [comment]
> @kalabYibeltal watch out, you might end up with merge conflicts because #52 is also working with the same README files. I think it would be good if you both work together.

Yes, I have seen that and since we are working together we are going to resolve that issue.

**Pull request closed without merging**